### PR TITLE
wxGUI/nviz: fix failed map centring on mac with pre-wx4.1

### LINKED ISSUE #590: [Bug] wxGUI nviz: map is no longer centered in the window
**Describe the bug**
When 3D view (nviz) is launched, the map is no longer centered in the window, but extends off the view to the upper right. It is very difficult to view all the map. Pan doesn't work until the tilt tool is used. 

**To Reproduce**
Open elevation map in NC demo data
Zoom to map to make sure it's all in the 2D view
Switch to 3D view
Map is mostly out of the view now

**Expected behavior**
Map should be centered in the view window when 3D mode is launched

**Screenshots**
nviz_screenshot

**System description (please complete the following information):**

- Mac OSX 10.14
- GRASS GIS version 7.8.3 downloaded and compiled 5 May 2020

### PR DESCRIPTION
**Pull request by nilason**
Fixes #590

Seems that the #590 issue with problem centring the map in nviz 3D-mode on hi-res displays, is an issue only for Mac with wxPython <4.1.0.

### CODE REVIEW

**Comment by nilason** [comment]
forgot: backport needed

**Comment by neteler** [comment]
You may simply set the label :)

**Comment by nilason** [comment]
> You may simply set the label :)

Didn't notice... thanks!

**Comment by nilason** [comment]
> GetContentScaleFactor is used in other places as well, so probably we should have a wrapper function for that.

Good point! If I'm not mistaken GLWindow.GetContentScaleFactor() is only used in that one file (nviz/mapwindow.py), overriding the method should be enough, wouldn't it? Or a new wrapper class? The latter seems to me a bit of an overkill.

**Comment by nilason** [comment]
This seems to work:
```
class GLWindow(MapWindowBase, glcanvas.GLCanvas):
...
    def GetContentScaleFactor(self):
        if sys.platform == "darwin" and not CheckWxVersion(version=[4, 1, 0]):
            return 1
        return super.GetContentScaleFactor()
```

**Comment by petrasovaa** [comment]
> This seems to work:
> 
> ```
> class GLWindow(MapWindowBase, glcanvas.GLCanvas):
> ...
>     def GetContentScaleFactor(self):
>         if sys.platform == "darwin" and not CheckWxVersion(version=[4, 1, 0]):
>             return 1
>         return super.GetContentScaleFactor()
> ```

Should this be super()? But otherwise seems ok.

**Comment by nilason** [comment]
Updated according to discussion.

File: gui/wxpython/nviz/mapwindow.py [hunk 1]
```diff
@@ -247,6 +247,11 @@ def _warningDepthBuffer(self):
                         "and restart GUI.")
             GMessage(message)
 
+    def GetContentScaleFactor(self):
+            if sys.platform == "darwin" and not CheckWxVersion(version=[4, 1, 0]):
+                return 1
+            return super().GetContentScaleFactor()
```
Comment (petrasovaa) [inline gui/wxpython/nviz/mapwindow.py:253]:
```suggestion
        if sys.platform == "darwin" and not CheckWxVersion(version=[4, 1, 0]):
            return 1
        return super().GetContentScaleFactor()
```

**Comment by nilason** [comment]
jee, thanks for the correction!

**Comment by nilason** [comment]
Thank you!

**Pull request merged**